### PR TITLE
[browser] Bump emscripten 2.0.14

### PR DESCRIPTION
**Pull request by kjpou1**
Active issue opened on emcripten : https://github.com/emscripten-core/emscripten/issues/13547
PR open here: https://github.com/emscripten-core/emscripten/pull/13549

Emscripten now builds a complete sysroot inside the EM_CACHE directory.  Introduced in 2.0.13
- This includes the system headers which get copied into place there rather than adding a sequence of extra include directories.

Add `-s TEXTDECODER=1` to EMCC_FLAGS to work around bug introduced 
* Default to TEXTDECODER=2 in -Oz builds to save 0.5KB-1KB of build output size. emscripten-core/emscripten#13304
* Fixes `TextDecoder is not defined` problem with running in console as well as when running the AOT console sample.

```
ReferenceError: TextDecoder is not defined
    at dotnet.js:1:8517
    at loadScript (runtime.js:196:3)
    at runtime.js:280:1

runtime.js:196: Error executing file
                load (url);
```

* Fixes `TextDecoder is not defined` problem when running tests in console.

```
info: ReferenceError: TextDecoder is not defined
  info:     at dotnet.js:1:8517
  info:     at loadScript (runtime.js:196:3)
  info:     at runtime.js:280:1
  info:
  info: runtime.js:196: Error executing file
  info:                 load (url);
  info:   ^
  info:
  info: 09:54:03.1461730 Process v8 exited with 1
```

### CODE REVIEW

**Comment by kjpou1** [comment]
Actually thinking this should be `TEXTDECODER=1` and not `TEXTDECODER=0`

File: src/mono/wasm/Makefile [hunk 1]
```diff
@@ -63,7 +63,7 @@ MONO_LIBS = \
 	$(ICU_LIBDIR)/libicuuc.a \
 	$(ICU_LIBDIR)/libicui18n.a
 
-EMCC_FLAGS=--profiling-funcs -s ALLOW_MEMORY_GROWTH=1 -s NO_EXIT_RUNTIME=1 -s USE_ZLIB=1  -s FORCE_FILESYSTEM=1 -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ccall', 'FS_createPath', 'FS_createDataFile', 'cwrap', 'setValue', 'getValue', 'UTF8ToString', 'UTF8ArrayToString', 'addFunction']" -s "EXPORTED_FUNCTIONS=['_putchar']" --source-map-base http://example.com  -emit-llvm
+EMCC_FLAGS=--profiling-funcs -s ALLOW_MEMORY_GROWTH=1 -s NO_EXIT_RUNTIME=1 -s USE_ZLIB=1 -s TEXTDECODER=1 -s FORCE_FILESYSTEM=1 -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ccall', 'FS_createPath', 'FS_createDataFile', 'cwrap', 'setValue', 'getValue', 'UTF8ToString', 'UTF8ArrayToString', 'addFunction']" -s "EXPORTED_FUNCTIONS=['_putchar']" --source-map-base http://example.com  -emit-llvm
```
Comment (akoeplinger) [inline src/mono/wasm/Makefile:66]: I think we should also set `-s AUTO_NATIVE_LIBRARIES=0` which was introduced in 2.0.13

**Review by akoeplinger — COMMENTED** [review]
looks good but we'll need to wait and bump to the new Docker image once it's available.

**Review by vargaz — APPROVED** [review]

File: src/mono/wasm/Makefile (hunk 1 above)
Comment (kjpou1) [reply src/mono/wasm/Makefile:66]: Looked at that as well but wanted to get all the other changes in first.  Will test that option and create a new PR with just that option.

**Comment by kjpou1** [comment]
Seems to be issues with `System.IO.FileSystem.CopyFile` with upgrade to 2.0.13 as well.

System.IO.FileNotFoundException : Unable to find the specified file.

Looking into the cause of change or report issue

**Comment by kjpou1** [comment]
Have been able to track the failure to [pal_io.c](https://github.com/dotnet/runtime/blob/master/src/libraries/Native/Unix/System.Native/pal_io.c#L1165) in the copying of the metadata.  

With 2.0.12 this returns `0` whereas  2.0.13 returns -1 and errno = 44.

This happened in 2.0.13 so that narrows it down.

**Comment by kjpou1** [comment]
This PR is the breaking change:  https://github.com/emscripten-core/emscripten/pull/13282

**Comment by kjpou1** [comment]
Created issue: https://github.com/emscripten-core/emscripten/issues/13547

**Pull request closed without merging**